### PR TITLE
feat: implement SDG in cluster for standalone mode

### PR DESCRIPTION
**Pull request by leseb**
e9fe873 feat: implement SDG in cluster for standalone mode

commit e9fe873eaefdff4b2d54494f922e6e25abfd69ed
Author: Sébastien Han <seb@redhat.com>
Date:   Tue Oct 22 12:24:01 2024 +0200

    feat: implement SDG in cluster for standalone mode
    
    Now, the standalone script supports running the SDG process in the
    end-to-end flow.
    
    Closes: https://github.com/redhat-et/ilab-on-ocp/issues/109
    Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
Using a Draft to indicate this should NOT merged. This should be considered once Phase 2 starts.

File: standalone/standalone.py [hunk 1]
```diff
@@ -2884,6 +2654,181 @@ def decode_base64(data):
...
+        1. Creates a PVC to hold SDG data and transformed SDG data.
+        2. Initiates a job to run a pod for SDG data preprocessing.
+    """
+    # Populate variables from context
+    namespace = ctx.obj["namespace"]
+    dry_run = ctx.obj["dry_run"]
+    force_pull = ctx.obj["force_pull"]
+    sdg_object_store_secret = ctx.obj["sdg_object_store_secret"]
+    sdg_serving_model_secret = ctx.obj["sdg_serving_model_secret"]
+    sdg_serving_model_secret = ctx.obj["sdg_serving_model_secret"]
```
Comment (HumairAK) [inline standalone/standalone.py:2677]: duplicate lines

File: standalone/standalone.py [hunk 2]
```diff
@@ -984,37 +1032,26 @@ def get_vol() -> list[kubernetes.client.V1Volume]:
...
-    num_instructions_to_generate: int,
-    exec_git_clone_op_repo_url: str = "",
-    exec_git_clone_op_repo_branch: str = "",
-    exec_git_clone_op_repo_pr: str = "",
-) -> kubernetes.client.V1Job:
+def create_sdg_container(
+    sdg_serving_model_secret: str,
+    num_instructions_to_generate: int = 30,
+    exec_git_clone_op_repo_branch: str = None,
+    exec_git_clone_op_repo_pr: str = None,
```
Comment (HumairAK) [inline standalone/standalone.py:1039]: these look like they are not used

**Comment by HumairAK** [comment]
@leseb just left some nits, otherwise this lgtm, I see that it's a draft currently, what else do we need to get this merged?

File: standalone/standalone.py (hunk 2 above)
Comment (leseb) [reply standalone/standalone.py:1039]: They are arguments to the `sdg_op` function. `exec_git_clone_op_repo_branch` and `exec_git_clone_op_repo_pr` are just not exposed to the CLI. I made a comment about that. Thanks!

**Comment by HumairAK** [comment]
@leseb the `make standalone` produces a diff here, iiuc it shouldn't, can you double check?

**Comment by leseb** [comment]
> @leseb the `make standalone` produces a diff here, iiuc it shouldn't, can you double check?

Correct, I updated the script and rebased.

**Pull request merged**